src/main/java/htsjdk/samtools/cram/structure/Slice.java

```diff
@@ -318,4 +333,113 @@ public CRAIEntry getCRAIEntry() {
     public CRAIEntry getCRAIEntry(final long containerStartOffset) {
         return new CRAIEntry(sequenceId, alignmentStart, alignmentSpan, containerStartOffset, offset, size);
     }
+
+    /**
+     * Create a Slice from CRAM Compression Records and a Compression Header
+     *
+     * @param records input CRAM Compression Records
+     * @param header the enclosing {@link Container}'s Compression Header
+     * @return a Slice corresponding to the given records
+     */
+    public static Slice buildSlice(final List<CramCompressionRecord> records,
```

Comment:
moved here from ContainerFactory.  May change this to a constructor in future Slice refactoring.

Reply:
Yes - the intention here is that this list of records all goes into the same slice.

`buildContainer()` handles dividing the list of records into slices, calling this once per result of that division.

I can add a comment about this.